src/utils/get-import-nodes-matched-group.ts

```diff
@@ -11,15 +15,43 @@ export const getImportNodesMatchedGroup = (
     node: ImportDeclaration,
     importOrder: string[],
 ) => {
-    const groupWithRegExp = importOrder.map((group) => ({
-        group,
-        regExp: new RegExp(group),
-    }));
+    const groupWithRegExp = importOrder
+        .sort((a, b) => {
+            if (
+                a.startsWith(TYPE_SPECIAL_WORD) &&
+                !b.startsWith(TYPE_SPECIAL_WORD)
+            ) {
+                return -1;
+            }
+            if (
+                !a.startsWith(TYPE_SPECIAL_WORD) &&
+                b.startsWith(TYPE_SPECIAL_WORD)
+            ) {
+                return 1;
+            }
+            return 0;
+        })
+        .map((group) => ({
+            group,
+            regExp: group.startsWith(TYPE_SPECIAL_WORD)
+                ? new RegExp(group.replace(TYPE_SPECIAL_WORD, ''))
+                : new RegExp(group),
+        }));
```

Comment:
Maybe extract this function to a separate file so you can test it?

Reply:
I've removed the sorting, not sure why I had it in the first place.